nd4j/samediff-import/samediff-import-onnx/src/main/kotlin/org/nd4j/samediff/frameworkimport/onnx/ir/OnnxIRGraphRunner.kt

```diff
@@ -50,14 +50,15 @@ class OnnxIRGraphRunner(graphDef: OnnxIRGraph, inputNames: List<String>, outputN
         //already in the model, we need to dynamically modify the model at runtime
         //to allow things like intermediate results
         outputNames.forEach {
-          graphDefBuilder.addOutput(ValueInfoProto {
-              name = it
-          })
+            if(!graphDef.outputList.contains(it))
```

Comment:
@treo this is just a simple check for detecting dups. We shiv in extra "nodes" to streamline the algorithm to work similar to TF import.